### PR TITLE
feat: extend ThemeData in MaterialApp

### PR DESCRIPTION
**Pull request by hoffmannkrzysztof**
Extend the theme in MaterialApp to add default styles for application. It prevent styling on some basic widgets.

### CODE REVIEW

**Comment by momshaddinury** [comment]
@hoffmannkrzysztof please update your branch with the latest changes from alpha. Resolve the conflicts.

**Comment by jorre127** [comment]
@hoffmannkrzysztof  Thanks for helping out. But there was a reason we were using the themeProvider haha. So you shouldn't remove all the theme providers. using the build in material theme certainly isn't a bad idea though as it gives a lot of ui items a standard color so you don't have to define it every time so you can definitely keep that. But when defining a custom color for a ui item it's better to use the custom implementation we currently have because it gives us way more freedom because it allows us to make our own ui tokens. Also the current implementation currently makes it easy to have multiple themes like dark mode, light mode, custom etc. Also i was planning to store the textThemes in the AppTheme as well so they also support multiple themes. So could you maybe undo all the places where you removed the theme provider, but keep the materialTheme you added? This will also help with the merge conflicts.

**Comment by hoffmannkrzysztof** [comment]
Sure no problem at all. I leave only changes in MaterialApp

**Comment by jorre127** [comment]
Thanks, sorry after you went trough the trouble.

**Review by jorre127 — APPROVED** [review]

File: lib/UI/workout/workout_page_widgets.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import 'package:flutter/material.dart';
+
+import 'workout_toatals_widget.dart';
+
+class WorkoutPageWidget extends StatelessWidget {
+  final bool firstLoad;
+  final WorkoutData workoutData;
+
+  const WorkoutPageWidget(
+      {Key? key, required this.firstLoad, required this.workoutData})
```
Comment (jorre127) [inline lib/UI/workout/workout_page_widgets.dart]: put a comma at the end and format

File: lib/UI/workout/workout_page_widgets.dart [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return ThemeProvider(builder: (context, theme) {
+      return GestureDetector(
+        onTap: () {
+          FocusScope.of(context).requestFocus(new FocusNode());
```
Comment (jorre127) [inline lib/UI/workout/workout_page_widgets.dart]: use arrow function here

File: lib/UI/workout/workout_page_widgets.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                          addAutomaticKeepAlives: true,
+                          children: workoutData.exerciseWidgets,
+                        ),
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+      );
+    });
```
Comment (jorre127) [inline lib/UI/workout/workout_page_widgets.dart]: add comma

File: lib/UI/workout/workout_page_widgets.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                color: theme.colorTheme.backgroundColorVariation,
+              )
+            : GestureDetector(
+                child: Container(
+                  color: theme.colorTheme.backgroundColorVariation,
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                    children: [
+                      WorkoutTotalsWidget(
+                        totals: workoutData.totals,
```
Comment (jorre127) [inline lib/UI/workout/workout_page_widgets.dart]: Personally when a widget only Hase 1 parameter I just put it on 1 line

File: lib/UI/workout/workout_page_widgets.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return ThemeProvider(builder: (context, theme) {
+      return GestureDetector(
+        onTap: () {
+          FocusScope.of(context).requestFocus(new FocusNode());
+        },
+        child: firstLoad
+            ? Container(
+                color: theme.colorTheme.backgroundColorVariation,
```
Comment (jorre127) [inline lib/UI/workout/workout_page_widgets.dart]: same here

**Comment by hoffmannkrzysztof** [comment]
Wrong branch :)

**Comment by momshaddinury** [comment]
> Wrong branch :)

What is the update of this?

**Comment by hoffmannkrzysztof** [comment]
I have already reverted unnecessary commit. So only changes to MaterialApp are included in this PR.

**Review by tenshiAMD — CHANGES_REQUESTED** [review]
@hoffmannkrzysztof please rebase and add a brief description in this PR. Thanks!

**Review by tenshiAMD — APPROVED** [review]

**Review by tenshiAMD — CHANGES_REQUESTED** [review]
@hoffmannkrzysztof LGTM however please add a brief description about your changes first before we can merge this one. Thanks!

**Review by tenshiAMD — APPROVED** [review]

**Comment by tenshiAMD** [comment]
@hoffmannkrzysztof LGTM, thanks! 

@dinurymomshad @jorre127 I will leave the merging for you guys

**Comment by tenshiAMD** [comment]
> @dinurymomshad @jorre127 I will leave the merging for you guys

@dinurymomshad @jorre127 gentle ping

**Pull request merged**